app/helpers/foreman_tasks/foreman_tasks_helper.rb

```diff
@@ -0,0 +1,190 @@
+        end
+      end
+    end
+
+    def datetime_f(f, attr, field_options = {}, datetime_options = {}, html_options = {})
+      f.fields_for attr do |fields|
+        field(fields, attr, field_options) do
+          fields.datetime_select attr, datetime_options, html_options
+        end
+      end
+    end
+
+    def inline_checkboxes_f(f, attr, field_options = {}, checkboxes = {}, options = {})
+      field(f, attr, field_options) do
+        checkboxes.map do |key, name|
+          [f.check_box(key, options), " #{name} "]
+        end.flatten.join('')
+      end
+    end
+
+    def trigger_selector(f, options = {})
+      tags = []
+      form_name = f.options[:html][:id]
+      script =<<-END
+          | $(function() { trigger_#{form_name}_selector_binds() });
+          |
+          | function trigger_#{form_name}_selector_binds() {
+          |   var form = $('form##{form_name}');
+          |   var trigger_mode_selector = form.find('input.trigger_mode_selector');
+          |   var input_type_selector = form.find('.form-control#input_type_selector');
+          |
+          |   trigger_mode_selector.on('click', function () {
+          |     ["future", "immediate", "recurring"].forEach(function(type) {
+          |       form.find('fieldset.trigger_mode_form#trigger_mode_' + type).hide();
+          |     });
+          |     form.find('fieldset.trigger_mode_form#trigger_mode_' + $(this).val()).show();
+          |   });
+          |
+          |   input_type_selector.on('change', function () {
+          |     var fieldset = form.find('fieldset.trigger_mode_form#trigger_mode_recurring');
+          |     ['cronline', 'monthly', 'weekly', 'hourly', 'daily'].forEach(function(type) {
+          |       fieldset.find('fieldset.input_type_form#input_type_' + type).hide();
+          |     })
+          |     fieldset.find('fieldset.input_type_form#input_type_' + $(this).val()).show();
+          |     var timepicker = fieldset.find('fieldset.input_type_form#time_picker');
+          |     if($(this).val() == 'cronline') {
+          |       timepicker.hide();
+          |     } else {
+          |       var hour_picker = timepicker.find('div.select2-container#s2id_#{f.object_name}_triggering_recurring_options_time_time_4i');
+          |       if($(this).val() == 'hourly') {
+          |         hour_picker.hide();
+          |       } else {
+          |         hour_picker.show();
+          |       }
+          |       timepicker.show();
+          |     }
+          |   });
+          |
+          |   form.find('input.end_time_limit_selector').on('click', function() {
+          |     var o = form.find('fieldset#trigger_mode_recurring fieldset#end_time_limit_form');
+          |     if($(this).val() === 'true') {
+          |       o.show();
+          |     } else {
+          |       o.hide();
+          |     };
+          |   });
+          | };
+      END
+      f.fields_for 'triggering' do |trigger_fields|
```

Comment:
What about extracting the html generation into the erb, and just rendering from here, as we do in foreman https://github.com/theforeman/foreman/blob/develop/app/helpers/application_helper.rb#L42